CYCLONEDDS_QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,150 @@
+This document is a declaration of software quality for the `CycloneDDS` external dependency imported by the rmw\_cyclonedds package, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+# `CycloneDDS` Quality Declaration
+
+The package `CycloneDDS` claims to be in the **Quality Level 4** category.
```

Comment:
Since this is an external dependency, it's tough to say that `CycloneDDS` is saying this. Maybe, "This QD claims that CycloneDDS ..."

Reply:
Does it help if I say that it is? It ought to!

Reply:
@eboasson claims `CycloneDDS` is in the **Quality Level 4** category. If you got a beef with it, talk to them!